### PR TITLE
Move MemoryStream to shared CoreLib partition

### PR DESCRIPTION
**Pull request by jkotas**
- Apply formating and other minor changes from CoreRT
   - Prefer async implementations from CoreCLR since they are more recent and better optimized
- Apply similar changes to UnmanagedMemoryStream as well to keep the two close

### CODE REVIEW

File: src/mscorlib/shared/System/IO/MemoryStream.cs [hunk 1]
```diff
@@ -235,15 +227,13 @@ internal byte[] InternalGetBuffer()
         // PERF: True cursor position, we don't need _origin for direct access
         internal int InternalGetPosition()
         {
-            if (!_isOpen) throw Error.GetStreamIsClosed();
```
Comment (stephentoub) [inline src/mscorlib/shared/System/IO/MemoryStream.cs]: It looks like InternalGetPosition is used by BinaryReader.  Does removing this have any negative impact there?

Comment (jkotas) [reply src/mscorlib/shared/System/IO/MemoryStream.cs]: The one place where it is called is immediately followed by a call to InternalEmulateRead that verifies that stream is not closed.

Comment (stephentoub) [reply src/mscorlib/shared/System/IO/MemoryStream.cs]: Cool

**Comment by jkotas** [comment]
@dotnet-bot test Windows_NT x64 corefx_baseline
@dotnet-bot test Ubuntu x64 corefx_baseline

**Pull request merged**